### DISCUSSION TITLE
Delay automerge PR comment until after reviews

### DISCUSSION THREAD

**Discussion by aaronjensen** [renovatebot/renovate#22168] [Suggest an Idea]
**What would you like Renovate to be able to do?**
Delay `automergeComment` until there is a review if a review is required or enable `renovate-approve` to leave the `automergeComment` instead.

**Describe the solution you'd like**
If https://github.com/renovatebot/renovate/issues/1853 were implemented and renovate could be configured to wait until the review status check was present, then that would effectively solve this, though it would likely cause a long delay between when the review happened and when the `automergeComment` happened.

Ideally, `renovate-approve` could be configured to leave the automerge message, or renovate could watch for approvals webhooks and be configured to expect that and not automerge until then.

**Describe alternatives you've considered**
Copying and modifying the `renovate-approve` application to issue the `bors r+` comment as part of the review.

**Additional context**
It's often fast enough, but not always (note that it actually looks like it made it in time, but it didn't:

![CleanShot 2019-12-20 at 11 57 13](https://user-images.githubusercontent.com/8588/71288552-ed467f00-231f-11ea-8f5b-918c986c3d58.png)

**Comment by roni-frantchi** [comment]
Hey @rarkins - any chance we'll see this one attended any time soon?..  
A configuration to only automerge **approved** PRs will probably do the trick

**Comment by rarkins** [comment]
@roni-frantchi this is low priority and without requirements completed so zero chance of being worked on.

If anyone wants to extend the existing approve bot functionality then the source is here: https://github.com/renovatebot/renovate-approve-bot and we should discuss requirements there too.

If we were to automerge only approved PRs then it would require some planning and research here:
- How to support old and new approaches simultaneously, e.g. with config option
- Most efficient way to determine approvals and for which platforms that is applicable
- Where in the logic flow that would occur
